packages/shylu/shylu_node/hts/src/shylu_hts_impl_def.hpp

```diff
@@ -4044,41 +4045,47 @@ ondiag_solve (const OnDiagTri& t, Sclr* x, const Int ldx, const Int nrhs,
   const Int nthreads = t.nthreads();
   if (nthreads == 1) {
     t.solve(x, ldx, x, ldx, nrhs);
+#ifdef _OPENMP
```

Comment:
Can you add a comment explaining why this is needed before the lock acquisition?

Reply:
One has to sync the local view of memory with the global view. Prior to this ARM architecture, this code ran only on Intel CPUs, and Intel CPUs tend to enforce memory consistency much more strongly than the OpenMP memory model requires. Thus, I never had a testbed to really get the flush placement tested robustly. ARM is very nifty in that it seems to take advantage of the OpenMP memory model quite efficiently, so it's great for flushing out any flush bugs.

Reply:
I added comments re: acquire and release flushes to the first two instances of each, to explain the general pattern.